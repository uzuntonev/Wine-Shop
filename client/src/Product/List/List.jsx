import React, { useState, useEffect } from 'react';
import {
  fetchPhotos,
  openUploadWidget,
} from '../../services/cloudinary-service';
import {
  Image,
  CloudinaryContext,
} from 'cloudinary-react';

const List = () => {
  const [images, setImages] = useState([]);

  const beginUpload = (tag) => {
    const uploadOptions = {
      cloudName: 'dfyamkucg',
      tags: [tag, 'anImage'],
      uploadPreset: 'upload',
    };
    openUploadWidget(uploadOptions, (error, photos) => {
      if (!error) {
        console.log(photos);
        if (photos.event === 'success') {
          setImages([...images, photos.info.public_id]);
        }
      } else {
        console.log(error);
      }
    });
  };

  useEffect(() => {
    fetchPhotos('image', setImages);
  }, []);
  return (
    <CloudinaryContext cloudName="dfyamkucg">
      <div className="App">
        <button onClick={() => beginUpload('image')}>Upload Image</button>
        <section>
          {images.map((i) => (
            <Image key={i} publicId={i} fetch-format="auto" quality="auto" />
          ))}
        </section>
      </div>
    </CloudinaryContext>
  );
};

export default List;
