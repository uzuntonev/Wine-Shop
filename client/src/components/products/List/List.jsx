import React, { useState, useEffect, useContext } from 'react';
import { CloudinaryContext } from 'cloudinary-react';
import { StoreContext } from '../../../Store/Store';
import { makeStyles } from '@material-ui/core/styles';
import productService from '../../../services/product-service';
import Card from '../Card/Card';
import { Grid } from '@material-ui/core';

const useStyles = makeStyles((theme) => {
  return {
    root: {
      padding: 20,
      alignSelf: 'center',
    },
    item: {
      margin: 20,
    },
  };
});
const List = () => {
  const classes = useStyles();
  const { state } = useContext(StoreContext);
  const [products, setProducts] = useState([]);
  useEffect(() => {
    if (state.isAuth) {
      productService.getUserProducts().then(({ data: products }) => {
        setProducts(products);
      });
    } else {
      // fetchPhotos('image', setImages);
      productService.getAllProducts().then(({ data }) => {
        setProducts(data);
      });
    }
  }, [state.isAuth]);

  const renderProducts = products.map((product) => {
    return (
      <Grid className={classes.item} item xs={3} key={product._id}>
        <Card product={product} />
      </Grid>
    );
  });
  return (
    <CloudinaryContext cloudName="dfyamkucg">
      <Grid container className={classes.root}>
        {renderProducts}
      </Grid>
    </CloudinaryContext>
  );
};

export default List;
