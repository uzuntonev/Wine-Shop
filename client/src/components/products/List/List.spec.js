import React from 'react';
import List from './List';
import renderer from 'react-test-renderer';
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import Context from '../../../store/store';

import { BrowserRouter } from 'react-router-dom';
test('Component "List" is mount', () => {
  const useContextSpy = jest
    .spyOn(React, 'useContext')
    .mockReturnValueOnce({ state: { products: [{},{}], isAuth: false } });
  const theme = createMuiTheme({
    palette: {
      primary: {
        main: '#121212',
        text: '#B0976D',
      },
    },
  });
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Context>
        <BrowserRouter>
          <List />
        </BrowserRouter>
      </Context>
    </ThemeProvider>
  );
  let tree = component.toJSON();
  expect(useContextSpy).toHaveBeenCalledTimes(6);
  expect(tree).toMatchSnapshot();
});
