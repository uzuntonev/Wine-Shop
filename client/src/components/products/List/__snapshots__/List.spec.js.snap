// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component "List" is mount 1`] = `
<div>
  <div
    className="MuiGrid-root-3 makeStyles-root-1 MuiGrid-container-4"
  />
</div>
`;
